### PR TITLE
Introduce updated locale picker

### PR DESCRIPTION
**Pull request by SiobhyB**
**_Note, this PR was originally created by @khaykov at https://github.com/wordpress-mobile/WordPress-Android/pull/14836. As the `develop` branch was deleted, it was not possible to reopen the original PR, which is why a new PR has been generated based on `trunk`._**

This PR updates the look of the app's locale picker, following the same design as the site timezone picker. It can be searched by both label and localized label.

[Image from Gyazo](https://gyazo.com/ef74e9dab13e3dbd1fab40555d8e5b04)

To test:
- Navigate to **App Settings**, and open the **Interface Language** option.
- Confirm that the new locale picker works as expected:
      - You can filter locales by both their label and localized label (eg. typing "Dani" and "Dans" should produce list with "Danish/Dansk" locale)
      - Tapping on locale changes app language.
      - Current app locale is displayed a the top of the list. Tapping on it closes the list.

## Regression Notes
1. Potential unintended areas of impact
- Locale Picker is not working!

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing + VM tests.

3. What automated tests I added (or what prevented me from doing so)
- Hopefully all of them!

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by SiobhyB** [comment]
@ravishanker, @develric, 👋 , I've requested you as reviewers as you'd originally reviewed the initial PR (and it seemed like it was close to being approved). Let me know if anything needs changing here or if there are perhaps other reviewers who should be added. I'd love to try to get this over the finish line. 🙇‍♀️ 

For further context, the reason I'm trying to get the improved locale picker merged is that I'd like to fix the following issue with missing language options in the site settings: https://github.com/wordpress-mobile/WordPress-Android/issues/16312. That fix will involve the addition of ~100 more language options under site settings, which wouldn't work very well with the existing locale picker.

**Review by khaykov — APPROVED** [review]
Works as expected! Thanks for updating it to work with trunk. I think we addressed all the issues in the original PR, and just never got to actually merging it.

**Comment by SiobhyB** [comment]
Thank you, @khaykov! 🙇‍♀️ @ravishanker, @develric, if you have no objections, I'll go ahead to merge this PR tomorrow.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    private fun LocalePickerBottomSheetBinding.setupObservers() {
+        viewModel.uiState.observe(viewLifecycleOwner, { uiState ->
+            uiState?.let {
+                localeAdapter.submitList(uiState.listData)
+                btnLocaleSuggestion.text = uiState.currentLocale?.label
+                emptyView.visibility = if (uiState.isEmptyViewVisible) View.VISIBLE else View.GONE
+            }
+        })
+
+        viewModel.hideKeyboard.observe(viewLifecycleOwner, {
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerBottomSheet.kt]: Lambda argument can be moved outside parenthesis like so.  Update all of them  
```
viewModel.hideKeyboard.observe(viewLifecycleOwner) {
        ActivityUtils.hideKeyboardForced(binding?.searchInputLayout)
}
```

**Review by ravishanker — APPROVED** [review]
Thanks for taking care of this.  🙇 
LGTM 👍

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerBottomSheet.kt (hunk 1 above)
Comment (SiobhyB) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerBottomSheet.kt]: Done in https://github.com/wordpress-mobile/WordPress-Android/pull/16331/commits/158d11e75258532d4125866344e655dca869e5da, thank you for reviewing! 🙌

**Pull request merged**